python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -131,17 +134,22 @@ def __init__(
             checkpoint_config=self._run_config.checkpoint_config,
             storage_context=self._storage_context,
         )
+        self._validation_manager = ValidationManager(
+            checkpoint_manager=self._checkpoint_manager,
+        )
         report_handler = ReportCallbackHandler(
             report_callbacks=(
-                [self._checkpoint_manager]
+                [self._checkpoint_manager, self._validation_manager]
                 + [c for c in self._callbacks if isinstance(c, ReportCallback)]
             )
         )
 
         # Group callbacks by the hooks they're subscribed to.
-        self._controller_callbacks = [self._scaling_policy] + [
-            c for c in self._callbacks if isinstance(c, ControllerCallback)
-        ]
+        self._controller_callbacks = (
+            [self._scaling_policy]
+            + [c for c in self._callbacks if isinstance(c, ControllerCallback)]
+            + [self._validation_manager]
```

Comment:
let's have validation manager come in front of potential user callbacks

```suggestion
            [self._scaling_policy, self._validation_manager]
            + [c for c in self._callbacks if isinstance(c, ControllerCallback)]
```

Reply:
done.